src/main/scala/com/mozilla/telemetry/pings/package.scala

```diff
@@ -159,6 +160,24 @@ package object pings {
       new Timestamp(this.Timestamp / 1000000)
     }
 
+    def normalizedBuildId(): Option[String] = {
+      `environment.build`.flatMap(_.buildId) match {
+        case Some(buildId: String) => {
+          val buildIdDay = buildId.slice(0, 8).toString()
+          val buildDateFormat = DateTimeFormat.forPattern("yyyyMMdd")
+          val buildDateTime = buildDateFormat.parseDateTime(buildIdDay)
```

Comment:
This will fail on improper dates which our schema validation [0] doesn't necessarily catch. For example, `00000000` is validated, but it will cause this to fail with `org.joda.time.IllegalFieldValueException: Cannot parse "00000000": Value 0 for monthOfYear must be in the range [1,12]`.

[0] https://github.com/mozilla-services/mozilla-pipeline-schemas/blob/master/schemas/telemetry/main/main.4.schema.json#L11